code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -78,6 +78,66 @@
 		message = pick("Oink!","Squeeeeeeee!","Oink Oink!")
 	return message
 
+///frog mask - reeee!!
+obj/item/clothing/mask/frog
+	name = "frog mask"
+	desc = "An ancient mask carved in the shape of a frog.<br> Sanity is like gravity, all it needs is a push."
+	icon_state = "frog"
+	item_state = "frog"
+	flags_inv = HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
+	w_class = WEIGHT_CLASS_SMALL
+	var/voicechange = TRUE
+	var/spam_flag = FALSE
+	var/delay = 15
+
+/obj/item/clothing/mask/frog/proc/reset_spam()
+	spam_flag = FALSE
+
+/obj/item/clothing/mask/frog/proc/play_ree()
+	playsound (src, 'sound/effects/reee.ogg', 30, 1)
+	spam_flag = TRUE
+	addtimer(CALLBACK(src, .proc/reset_spam), delay)

```

Comment:
rather than a timer have
`last_sound = world.time`
then in the play procs check
`if(world.time + delay > last_sound)`